onecodex/distance.py

```diff
@@ -28,9 +29,18 @@ def alpha_diversity(self, metric=AlphaDiversityMetric.Shannon, rank=Rank.Auto):
                 )
             )
 
+        if metric == "chao1":
+            warnings.warn(
+                "`Chao1` is deprecated and will be removed in a future release. Please use `observed_taxa` instead.",
+                DeprecationWarning,
+            )
+
         df = self.to_df(rank=rank, normalize=self._guess_normalized())
 
-        output = skbio.diversity.alpha_diversity(metric, df.values, df.index, validate=False)
+        if metric == "observed_taxa":
+            output = df[df.columns].gt(0).sum(axis=1)
```

Comment:
It might be better to pass `metric="observed_otus"` to scikit-bio's `alpha_diversity` function, just to avoid reimplementing that metric here. We could still use the name "observed_taxa" in our output. The metric is super simple though, so feel free to ignore this comment.

Here's what that might look like:

```python
skbio_metric = "observed_otus" if metric == "observed_taxa" else metric
output = skbio.diversity.alpha_diversity(skbio_metric, df.values, df.index, validate=False)
```

Side note: I like the name "observed_taxa" much better than "observed_species" or "observed_otus" 🎉  -- this has always been a confusing/historical leftover from qiime1 days 😄

Reply:
@lynchde sorry for flip-flopping - I know I told you the opposite before but I like @jairideout's solution here.

Reply:
Implemented @jairideout 's suggestion.